tests/manage/pv_services/test_pvc_rwx_writeable_after_pod_deletions.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import pytest
+
+from concurrent.futures import ThreadPoolExecutor
+from ocs_ci.framework.testlib import ManageTest, tier1
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from tests import helpers
+
+logger = logging.getLogger(__name__)
+
+
+def verify_pod_node(pod_obj, node_name):
+    """
+    Verifies that the pod is running on a particular node
+
+    Args:
+        pod_obj (Pod): The pod object
+        node_name (str): The name of node to check
+
+    Returns:
+        bool: True if the pod is running on a particular node, False otherwise
+    """
+
+    logger.info(f"Checking whether the pod is running on node: {node_name}")
+    actual_node = pod_obj.get().get('spec').get('nodeName')
+    logger.info(f"The pod is running on node: {actual_node}")
+    if actual_node == node_name:
+        return True
+    else:
+        return False
+
+
+class TestRWXMountPoint(ManageTest):
+    """
+    Automates the following test cases:
+    OCS-965	CEPHFS RWX: While two app pods access same mount,
+        delete one pod. Confirm second pod is still able to write
+    """
+
+    @tier1
+    @pytest.mark.parametrize(
+        argnames=["interface"],
+        argvalues=[
+            pytest.param(
+                *[constants.CEPHFILESYSTEM],
+                marks=pytest.mark.polarion_id("OCS-965")
+            )
+        ]
+    )
+    def test_pvc_rwx_writeable_after_pod_deletions(
+        self, interface, pvc_factory, teardown_factory
+    ):
+        """
+        Test assign nodeName to a pod using RWX pvc
+
+        1. Create a new project 	A new project is created.
+        2. Create a RWX CEPHFS based PVC
+        3. Attach the same PVC to multiple PODs and start IO on all the PODs
+        4. Delete all but one pod.
+        5. Verify mount point is still write-able.
+             - Start IO again on the Running pod.
+        6. Also, access the data written by deleted pods from the Running pod
+
+        """
+        worker_nodes_list = helpers.get_worker_nodes()
+
+        # Create a RWX PVC
+        pvc_obj = pvc_factory(
+            interface=interface, access_mode=constants.ACCESS_MODE_RWX,
+            size=10, status=constants.STATUS_BOUND
+        )
+        logger.info(
+            f"Creating pods on all worker nodes backed"
+            f"with same pvc {pvc_obj.name}"
+        )
+
+        pod_list = [helpers.create_pod(
```

Comment:
Please avoid using list comprehension here.
The reason is that in case one of the pods creation will fail, pod_list won't get filled with the ones that got created, and won't be deleted on teardown

Reply:
instead, append to a regular list and in each iteration, call teardown_factory(pod_obj)

Reply:
Done

```
pod_list = []

        for each_node in worker_nodes_list:
            pod_obj = helpers.create_pod(
                interface_type=interface, pvc_name=pvc_obj.name,
                namespace=pvc_obj.namespace, node_name=each_node,
                pod_dict_path=constants.NGINX_POD_YAML
            )
            pod_list.append(pod_obj)
            teardown_factory(pod_obj)
```

Please review